x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -99,6 +108,58 @@ func parse(b []byte) ([]statsdMetric, error) {
 	return metrics, nil
 }
 
+func newMetricFilter(mappings []StatsdMapping) *metricFilter {
+	return &metricFilter{
+		mappings: mappings,
+	}
+}
+
+func eventMapping(metricName string, metricValue interface{}, metricSetFields common.MapStr, mappings []StatsdMapping) {
+	if len(mappings) == 0 {
+		metricSetFields[common.DeDot(metricName)] = metricValue
+		return
+	}
+
+	for _, mapping := range mappings {
+		// The metricname match the one with no labels in mappings
+		// Let's insert it dedotted and continue
+		if metricName == mapping.Metric {
+			metricSetFields[mapping.Value.Field] = metricValue
+			return
+		}
+
+		regexPattern := strings.Replace(mapping.Metric, ".", `\.`, -1)
+		regexPattern = strings.Replace(regexPattern, "<", "(?P<", -1)
+		regexPattern = strings.Replace(regexPattern, ">", ">[^.]+)", -1)
+		r := regexp.MustCompile(fmt.Sprintf("^%s$", regexPattern))
```

Comment:
Not sure how much this can affect performance, but would it make sense to pre-compile the regexp for a given mapping so they don't need to be compiled every time `eventMapping` is called?

In any case I think that this needs to be pre-compiled at least once during initialization to check the validity of the user input. This regexp is built from the config and it could be invalid for example if a user forgets a `>`. In that case `MustCompile` would cause a panic, what is not nice.

Reply:
👍